---
title: "Appendices: R code"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# Load data ---------------------------------------------------------------

# Temporal veg
notree.all <- read.csv("../data/publish/Herb-and-shrub-cover_2012-2021.csv")
herb.all <- read.csv("../data/publish/Herb-cover_2012-2021.csv") 
shrub.all <- read.csv("../data/publish/Shrub-cover_2012-2021.csv")
invasive.all <- read.csv("../data/publish/Invasive-cover_2012-2021.csv")
plant.all <- read.csv("../data/publish/Species-cover_2012-2021.csv")
per.div <- read.csv("../data/publish/Perennial-plant-diversity_2012-2021.csv")

# 2021 data
dat.2021 <- read.csv("../data/publish/Veg-soil-elev_2021.csv")

# Sequencing
barc.asv <- read.table("../data/publish/bac-arc_clean_asv.txt", sep = "\t",
                       header = T, row.names = 1)
fungi.asv <- read.table("../data/publish/fungi_clean_asv.txt",
                        sep = "\t", header = T, row.names = 1)
meta <- read.csv("../data/publish/sequencing_metadata.csv")
```

Appendices for "Dryland rock detention structures increase herbaceous vegetation cover and stabilize shrub cover over 10 years but do not directly affect soil fertility" by Ossanna et al. (2023). Included are code and analysis used to make figures and full model output.  
  
Contact: Lia Ossanna, <lossanna@arizona.edu>  
Last updated: 2023-09-01  
\
\
\

Appendices: 

- Appendix A: Temporal vegetation trends, Figs 2 & S4
- Appendix B: Coefficient of variation, Figs 3 & S5
- Appendix C: Soil fertility, Figs S6-S8
- Appendix D: Structural equation modeling, models 1-4
\
\

Package versions:

- `R version 4.3.1`
- `tidyverse 2.0.0`
  - `dplyr 1.1.2`
  - `stringr 1.5.0`
  - `ggplot 3.4.3`
  - `tidyr 3.2.1`
- `agricolae 1.3-6`
- `plotrix 3.8-2`
- `ggpubr 0.6.0`
- `rstatix 0.7.2`
- `scales 1.2.1`
- `metagenomeSeq 1.42.0`, `Biobase 2.60.0`, `BiocGenerics 0.46.0`
- `vegan 2.6-4`
- `dada2 1.28.0`, `ShortRead 1.58.0`, `Biostrings 2.68.1` used for DADA2 pipeline (code not included here)
- `lavaan 0.6-16`


\newpage
# Appendix A: Temporal vegetation trends
## Setup
```{r, message=FALSE}
library(tidyverse)
library(agricolae)
library(plotrix)
library(ggpubr)
library(rstatix)
```
```{r, eval=FALSE}
# Load data ---------------------------------------------------------------

notree.all <- read.csv("Herb-and-shrub-cover_2012-2021.csv")
herb.all <- read.csv("Herb-cover_2012-2021.csv") 
shrub.all <- read.csv("Shrub-cover_2012-2021.csv")
invasive.all <- read.csv("Invasive-cover_2012-2021.csv")
plant.all <- read.csv("Species-cover_2012-2021.csv")
per.div <- read.csv("Perennial-plant-diversity_2012-2021.csv")
```
```{r}
# Functions ---------------------------------------------------------------

# Convert columns to factor or date as needed
convert.cols <- function(x) {
  x$year.xaxis <- as.Date(x$year.xaxis)
  
  group.cols <- c("Sample", "Year", "Treatment")
  
  x[group.cols] <- lapply(x[group.cols], factor)
  
  return(x)
}


# Data wrangling ----------------------------------------------------------

notree.all <- convert.cols(notree.all)
herb.all <- convert.cols(herb.all)
shrub.all <- convert.cols(shrub.all)
invasive.all <- convert.cols(invasive.all)
per.div <- convert.cols(per.div)
```

## Fig 2a: Grass, forb & shrub cover
```{r, fig.height=3, fig.width=6}
# Grass, forb, and shrub cover (notree) -----------------------------------

# Find averages by year
notree.avg <- notree.all %>% 
  group_by(Treatment, Year, year.xaxis) %>% 
  summarise(mean = mean(Cover),
            SD = sd(Cover),
            SE = std.error(Cover),
            .groups = "keep")

# One-way ANOVA for Control
summary(aov(Cover ~ Year, data = filter(notree.all, Treatment == "Control"))) # p = 4.3e-06
notree.ctrl <- notree.all |> 
  filter(Treatment == "Control")
anova.notree.ctrl <- aov(notree.ctrl$Cover ~ notree.ctrl$Year)
hsd.notree.ctrl <- HSD.test(anova.notree.ctrl, trt = "notree.ctrl$Year")
hsd.notree.ctrl$groups

# One-way ANOVA for Treated
summary(aov(Cover ~ Year, data = filter(notree.all, Treatment == "Treated"))) # p = 0.00304
notree.trt <- notree.all |> 
  filter(Treatment == "Treated")
anova.notree.trt <- aov(notree.trt$Cover ~ notree.trt$Year)
hsd.notree.trt <- HSD.test(anova.notree.trt, trt = "notree.trt$Year")
hsd.notree.trt$groups

# Plot with one-way ANOVA letters
notree.ctrl.letters <- hsd.notree.ctrl$groups
notree.ctrl.letters <- notree.ctrl.letters |> 
  mutate(Year = rownames(notree.ctrl.letters)) |> 
  arrange(Year)
notree.trt.letters <- hsd.notree.trt$groups
notree.trt.letters <- notree.trt.letters |> 
  mutate(Year = rownames(notree.trt.letters)) |> 
  arrange(Year)

letters.notree <- data.frame(x = rep(notree.avg$year.xaxis[1:6], 2),
                             y = rep(49, 12),
                             label = c(notree.ctrl.letters$groups,
                                       notree.trt.letters$groups),
                             Treatment = c(rep("Control", 6),
                                           rep("Treated", 6)))
ptext.notree <- data.frame(x = rep(as.Date("2019-09-01"), 2),
                           y = c(22, 22),
                           label = c("ANOVA, p < 0.001", "ANOVA, p = 0.003"),
                           Treatment = c("Control", "Treated"))

notree.plot <- ggplot(notree.avg, aes(x = year.xaxis, y = mean, 
                                      group = Treatment, 
                                      color = Treatment)) +
  geom_line() +
  geom_point() +
  geom_pointrange(aes(ymin = mean - SE, ymax = mean + SE)) +
  facet_wrap(~Treatment) +
  xlab(NULL) +
  ylab("Cover (%)") +
  ggtitle("Vegetation cover, 2012-2021") +
  scale_color_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  theme(legend.position = "none") +
  geom_text(data = letters.notree,
            mapping = aes(x = x, y = y, label = label),
            color = "black",
            size = 3.5)  +
  geom_text(data = ptext.notree,
            aes(x = x, y = y, label = label),
            color = "gray30",
            size = 2.5) +
  theme(axis.text.x = element_text(color = "black")) +
  theme(plot.margin = margin(t = 0.1, r = 0.1, b = 0.2, l = 0.1, "in")) 
notree.plot
```

## Fig 2b: Herbaceous (grass & forb) cover
```{r, fig.height=3, fig.width=6}
# Herbaceous cover --------------------------------------------------------

# Find averages by year
herb.avg <- herb.all %>% 
  group_by(Treatment, Year, year.xaxis) %>% 
  summarise(mean = mean(Cover),
            SD = sd(Cover),
            SE = std.error(Cover),
            .groups = "keep")

# One-way ANOVA for Control
summary(aov(Cover ~ Year, data = filter(herb.all, Treatment == "Control"))) # 0.00434
herb.ctrl <- herb.all |> 
  filter(Treatment == "Control")
anova.herb.ctrl <- aov(herb.ctrl$Cover ~ herb.ctrl$Year)
hsd.herb.ctrl <- HSD.test(anova.herb.ctrl, trt = "herb.ctrl$Year")
hsd.herb.ctrl$groups

# One-way ANOVA for Treated
summary(aov(Cover ~ Year, data = filter(herb.all, Treatment == "Treated"))) # p = 3.77e-10
herb.trt <- herb.all |> 
  filter(Treatment == "Treated")
anova.herb.trt <- aov(herb.trt$Cover ~ herb.trt$Year)
hsd.herb.trt <- HSD.test(anova.herb.trt, trt = "herb.trt$Year")
hsd.herb.trt$groups

# Plot with one-way ANOVA letters
herb.ctrl.letters <- hsd.herb.ctrl$groups
herb.ctrl.letters <- herb.ctrl.letters |> 
  mutate(Year = rownames(herb.ctrl.letters)) |> 
  arrange(Year)
herb.trt.letters <- hsd.herb.trt$groups
herb.trt.letters <-herb.trt.letters |> 
  mutate(Year = rownames(herb.trt.letters)) |> 
  arrange(Year)

letters.herb <- data.frame(x = rep(herb.avg$year.xaxis[1:6], 2),
                           y = rep(32, 12),
                           label = c(herb.ctrl.letters$groups,
                                     herb.trt.letters$groups),
                           Treatment = c(rep("Control", 6),
                                         rep("Treated", 6)))
ptext.herb <- data.frame(x = rep(as.Date("2019-09-01"), 2),
                         y = c(8, 8),
                         label = c("ANOVA, p = 0.004", "ANOVA, p < 0.001"),
                         Treatment = c("Control", "Treated"))

herb.plot <- ggplot(herb.avg, aes(x = year.xaxis, y = mean, 
                                  group = Treatment, 
                                  color = Treatment)) +
  geom_line() +
  geom_point() +
  geom_pointrange(aes(ymin = mean - SE, ymax = mean + SE)) +
  facet_wrap(~Treatment) +
  xlab(NULL) +
  ylab("Cover (%)") +
  ggtitle("Herbaceous cover") +
  scale_color_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  theme(legend.position = "none") +
  geom_text(data = letters.herb,
            mapping = aes(x = x, y = y, label = label),
            color = "black",
            size = 3.5)  +
  geom_text(data = ptext.herb,
            aes(x = x, y = y, label = label),
            color = "gray30",
            size = 2.5) +
  theme(axis.text.x = element_text(color = "black")) +
  theme(plot.margin = margin(t = 0.1, r = 0.1, b = 0.2, l = 0.1, "in")) 
herb.plot
```

## Fig 2c: Shrub cover
```{r, fig.height=3, fig.width=6}
# Shrub cover -------------------------------------------------------------

# Find averages by year
shrub.avg <- shrub.all %>% 
  group_by(Treatment, Year, year.xaxis) %>% 
  summarise(mean = mean(Cover),
            SD = sd(Cover),
            SE = std.error(Cover),
            .groups = "keep")

# One-way ANOVA for Control
summary(aov(Cover ~ Year, data = filter(shrub.all, Treatment == "Control"))) # p = 0.0112
shrub.ctrl <- shrub.all |> 
  filter(Treatment == "Control")
anova.shrub.ctrl <- aov(shrub.ctrl$Cover ~ shrub.ctrl$Year)
hsd.shrub.ctrl <- HSD.test(anova.shrub.ctrl, trt = "shrub.ctrl$Year")
hsd.shrub.ctrl$groups

# One-way ANOVA for Treated
summary(aov(Cover ~ Year, data = filter(shrub.all, Treatment == "Treated"))) # NS, p = 0.982


# Plot with one-way ANOVA letters
shrub.ctrl.letters <- hsd.shrub.ctrl$groups
shrub.ctrl.letters <- shrub.ctrl.letters |> 
  mutate(Year = rownames(shrub.ctrl.letters)) |> 
  arrange(Year)

letters.shrub <- data.frame(x = shrub.avg$year.xaxis[1:6],
                            y = rep(21, 6),
                            label = shrub.ctrl.letters$groups,
                            Treatment = rep("Control", 6))
ptext.shrub <- data.frame(x = rep(as.Date("2019-09-01"), 2),
                          y = c(6.5, 6.5),
                          label = c("ANOVA, p = 0.011", "ANOVA, p = 0.982"),
                          Treatment = c("Control", "Treated"))

shrub.plot <- ggplot(shrub.avg, aes(x = year.xaxis, y = mean, 
                                    group = Treatment, 
                                    color = Treatment)) +
  geom_line() +
  geom_point() +
  geom_pointrange(aes(ymin = mean - SE, ymax = mean + SE)) +
  facet_wrap(~Treatment) +
  xlab(NULL) +
  ylab("Cover (%)") +
  ggtitle("Shrub cover") +
  scale_color_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  theme(legend.position = "none") +
  geom_text(data = letters.shrub,
            mapping = aes(x = x, y = y, label = label),
            color = "black",
            size = 3.5)  +
  geom_text(data = ptext.shrub,
            aes(x = x, y = y, label = label),
            color = "gray30",
            size = 2.5) +
  theme(axis.text.x = element_text(color = "black")) +
  theme(plot.margin = margin(t = 0.1, r = 0.1, b = 0.2, l = 0.1, "in")) 
shrub.plot
```

## Combine plots for Fig 2
```{r, eval=FALSE}
# Combine notree, herb & shrub --------------------------------------------

tiff("figures/2023-09_publish-figures/temporal-ANOVA_notree-herb-shrub.tiff", units = "in", height = 8, width = 6, res = 1000)
ggarrange(notree.plot, herb.plot, shrub.plot,
          ncol = 1, nrow = 3,
          labels = c("(A)", "(B)", "(C)")) 

dev.off()
```



## Supp Fig 4a: Perennial plant species richness
```{r, fig.height=3, fig.width=6}
# Perennial plant richness ------------------------------------------------

# Find averages by year
rich.avg <- per.div %>% 
  group_by(Treatment, Year, year.xaxis) %>% 
  summarise(mean = mean(rich),
            SD = sd(rich),
            SE = std.error(rich),
            .groups = "keep")

# One-way ANOVA for Control
summary(aov(rich ~ Year, data = filter(per.div, Treatment == "Control"))) # 0.00881
rich.ctrl <- per.div |> 
  filter(Treatment == "Control")
anova.rich.ctrl <- aov(rich.ctrl$rich ~ rich.ctrl$Year)
hsd.rich.ctrl <- HSD.test(anova.rich.ctrl, trt = "rich.ctrl$Year")
hsd.rich.ctrl$groups

# One-way ANOVA for Treated
summary(aov(rich ~ Year, data = filter(per.div, Treatment == "Treated"))) # p = 0.0516

# Plot with one-way ANOVA letters
rich.ctrl.letters <- hsd.rich.ctrl$groups
rich.ctrl.letters <- rich.ctrl.letters |> 
  mutate(Year = rownames(rich.ctrl.letters)) |> 
  arrange(Year)

letters.rich <- data.frame(x = rich.avg$year.xaxis[1:6],
                           y = rep(10.4, 6),
                           label = rich.ctrl.letters$groups,
                           Treatment = rep("Control", 6))
ptext.rich <- data.frame(x = rep(as.Date("2019-09-01"), 2),
                         y = c(6.7, 10.3),
                         label = c("ANOVA, p = 0.009", "ANOVA, p = 0.052"),
                         Treatment = c("Control", "Treated"))

rich.plot <- ggplot(rich.avg, aes(x = year.xaxis, y = mean, 
                                  group = Treatment, 
                                  color = Treatment)) +
  geom_line() +
  geom_point() +
  geom_pointrange(aes(ymin = mean - SE, ymax = mean + SE)) +
  facet_wrap(~Treatment) +
  xlab(NULL) +
  ylab("No. of species") +
  ggtitle("Perennial plant species richness, 2012-2021") +
  scale_color_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  theme(legend.position = "none") +
  geom_text(data = letters.rich,
            mapping = aes(x = x, y = y, label = label),
            color = "black",
            size = 3.5) +
  geom_text(data = ptext.rich,
            aes(x = x, y = y, label = label),
            color = "gray30",
            size = 2.5) +
  theme(axis.text.x = element_text(color = "black")) +
  theme(plot.margin = margin(0.1, 0.1, 0.2, 0.1, "in")) 
rich.plot
```


## Supp Fig 4b: Perennial plant diversity
```{r, fig.height=3, fig.width=6}
# Perennial plant diversity (Shannon) -------------------------------------

# Find averages by year
shan.avg <- per.div %>% 
  group_by(Treatment, Year, year.xaxis) %>% 
  summarise(mean = mean(shan),
            SD = sd(shan),
            SE = std.error(shan),
            .groups = "keep")

# One-way ANOVA for Control
summary(aov(shan ~ Year, data = filter(per.div, Treatment == "Control"))) # p = 0.934

# One-way ANOVA for Treated
summary(aov(shan ~ Year, data = filter(per.div, Treatment == "Treated"))) # p = 0.725


# Plot with one-way ANOVA
ptext.shan <- data.frame(x = rep(as.Date("2019-09-01"), 2),
                         y = c(1.33, 1.33),
                         label = c("ANOVA, p = 0.934", "ANOVA, p = 0.725"),
                         Treatment = c("Control", "Treated"))

shan.plot <- ggplot(shan.avg, aes(x = year.xaxis, y = mean, 
                                  group = Treatment, 
                                  color = Treatment)) +
  geom_line() +
  geom_point() +
  geom_pointrange(aes(ymin = mean - SE, ymax = mean + SE)) +
  facet_wrap(~Treatment) +
  xlab(NULL) +
  ylab("Shannon diversity index") +
  ggtitle("Perennial plant diversity") +
  scale_color_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(color = "black")) +
  geom_text(data = ptext.shan,
            aes(x = x, y = y, label = label),
            color = "gray30",
            size = 2.5) +
  theme(plot.margin = margin(0.1, 0.1, 0.2, 0.1, "in")) 
shan.plot
```

## Combine plots for Supp Fig 4
```{r, eval=FALSE}
# Combine richness & Shannon ----------------------------------------------

tiff("figures/2023-09_publish-figures/temporal-ANOVA_richness-Shannon.tiff", units = "in", height = 5.5, width = 6, res = 300)
ggarrange( rich.plot, shan.plot,
           ncol = 1, nrow = 2,
           labels = c("(A)", "(B)")) 

dev.off()
```


## Invasive cover and most common species
```{r}
# Average cover by year
invasive.all %>% 
  select(-year.xaxis) |> 
  group_by(Treatment, Year) %>% 
  summarise(mean = mean(Cover),
            SE = std.error(Cover),
            .groups = "keep") |> 
  mutate_if(is.numeric, round, digits = 2)

# Most common species in Control
plant.all |> 
  filter(Native == "Invasive",
         Treatment == "Control") |> 
  group_by(Common) |> 
  summarise(mean = mean(Cover)) |> 
  arrange(desc(mean))

# Most common species in Treated
plant.all |> 
  filter(Native == "Invasive",
         Treatment == "Treated") |> 
  group_by(Common) |> 
  summarise(mean = mean(Cover)) |> 
  arrange(desc(mean)) 
```



\newpage
# Appendix B: Coefficient of variation
## Setup
```{r, message=FALSE}
library(tidyverse)
library(car)
library(scales)
library(ggpubr)
```
```{r, eval=FALSE}
# Load data ---------------------------------------------------------------

notree.all <- read_csv("Herb-and-shrub-cover_2012-2021.csv")
herb.all <- read_csv("Herb-cover_2012-2021.csv") 
shrub.all <- read_csv("Shrub-cover_2012-2021.csv")
per.div <- read_csv("Perennial-plant-diversity_2012-2021.csv")
```

## Fig 3a: CV of shrub cover
```{r, fig.height=4, fig.width=3}
# Shrub cover -------------------------------------------------------------

# Find CV for each sample over time
shrub.sample <- shrub.all |> 
  group_by(Sample, Treatment) |> 
  summarise(CV = sd(Cover) / mean(Cover),
            .groups = "keep") # NaNs produced because some have 0 cover and can't divide by 0

# Replace NaNs with 0
shrub.sample[1, 3] <- 0
shrub.sample[5, 3] <- 0
shrub.sample[8, 3] <- 0


# Compare means
wilcox.test(filter(shrub.sample, Treatment == "Treated")$CV,
            filter(shrub.sample, Treatment == "Control")$CV,
            exact = FALSE) # p = 0.01429


# Plot
letters.shrub <- data.frame(x = c(1, 2),
                            y = c(2.5, 2.5),
                            label = c("a", "b"))

shrub.plot.cv <- shrub.sample |> 
  ggplot(aes(x = Treatment, y = CV)) +
  geom_boxplot(alpha = 0.3,
               outlier.shape = NA,
               aes(fill = Treatment)) +
  geom_jitter(size = 1,
              alpha = 0.8,
              aes(color = Treatment)) +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  labs(title = "Shrub cover",
       x = NULL,
       y = "Coefficient of variation") +
  theme_bw() +
  theme(legend.position = "none") +
  scale_y_continuous(labels = percent)  +
  theme(axis.text.x = element_text(color = "black")) +
  geom_text(aes(x = 0.95, y = 2.75, label = "Mann-Whitney, \np = 0.014"),
            color = "gray30",
            size = 2.5) +
  geom_text(data = letters.shrub,
            aes(x = x, y = y, label = label),
            color = "black") +
  theme(plot.margin = margin(0.1, 0, 0.1, 0.1, "in")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
shrub.plot.cv
```


## Fig 3b: CV of herbaceous cover
```{r, fig.height=4, fig.width=3}
# Herb cover --------------------------------------------------------------

# Find CV for each sample over time
herb.sample <- herb.all |> 
  group_by(Sample, Treatment) |> 
  summarise(CV = sd(Cover) / mean(Cover),
            .groups = "keep")


# Compare means
wilcox.test(filter(herb.sample, Treatment == "Treated")$CV,
            filter(herb.sample, Treatment == "Control")$CV) # NS, p = 0.148

# Plot
herb.plot.cv <- herb.sample |> 
  ggplot(aes(x = Treatment, y = CV)) +
  geom_boxplot(alpha = 0.3,
               outlier.shape = NA,
               aes(fill = Treatment)) +
  geom_jitter(size = 1,
              alpha = 0.8,
              aes(color = Treatment)) +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  labs(title = "Herbaceous cover",
       x = NULL, 
       y = NULL) +
  theme_bw() +
  theme(legend.position = "none") +
  scale_y_continuous(labels = percent)  +
  theme(axis.text.x = element_text(color = "black")) +
  geom_text(aes(x = 0.95, y = 1.18, label = "Mann-Whitney, \np = 0.122"),
            color = "gray30",
            size = 2.5) +
  theme(plot.margin = margin(0.1, 0.1, 0.1, 0.2, "in")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
herb.plot.cv
```


## Fig 3c: CV of overall veg cover
```{r, fig.height=4, fig.width=3}
# Notree cover ------------------------------------------------------------

# Find CV for each sample over time
notree.sample <- notree.all |> 
  group_by(Sample, Treatment) |> 
  summarise(CV = sd(Cover) / mean(Cover),
            .groups = "keep")

# Compare means
t.test(filter(notree.sample, Treatment == "Treated")$CV,
       filter(notree.sample, Treatment == "Control")$CV) # NS, p = 0.882

# Plot
notree.plot.cv <- notree.sample |> 
  ggplot(aes(x = Treatment, y = CV)) +
  geom_boxplot(alpha = 0.3,
               outlier.shape = NA,
               aes(fill = Treatment)) +
  geom_jitter(size = 1,
              alpha = 0.8,
              aes(color = Treatment)) +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  labs(title = "Vegetation cover",
       x = NULL,
       y = NULL) +
  theme_bw() +
  theme(legend.position = "none") +
  scale_y_continuous(labels = percent) +
  theme(axis.text.x = element_text(color = "black")) +
  geom_text(aes(x = 0.9, y = 0.95, label = "t-test, p = 0.882"),
            color = "gray30",
            size = 2.5) +
  theme(plot.margin = margin(0.1, 0.1, 0.1, 0.2, "in")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
notree.plot.cv
```

## Combine plots for Fig 3
```{r, eval=FALSE}
# Combine notree, herb, shrub ---------------------------------------------

tiff("figures/2023-09_publish-figures/CV_shrub-herb-notree.tiff", units = "in", height = 4, width = 7, res = 1000)
ggarrange(shrub.plot.cv, herb.plot.cv, notree.plot.cv,
          ncol = 3, nrow = 1,
          labels = c("(A)", "(B)", "(C)")) 

dev.off()
```


## Supp Fig 5a: CV of perennial richness
```{r, fig.height=4, fig.width=3}
# Richness ----------------------------------------------------------------

# Find CV for each sample over time
rich.sample <- per.div |> 
  group_by(Sample, Treatment) |> 
  summarise(CV = sd(rich) / mean(rich),
            .groups = "keep")

# Compare means
wilcox.test(filter(rich.sample, Treatment == "Treated")$CV,
            filter(rich.sample, Treatment == "Control")$CV) # NS, p = 0.093

# Plot
rich.plot.cv <- rich.sample |> 
  ggplot(aes(x = Treatment, y = CV)) +
  geom_boxplot(alpha = 0.3,
               outlier.shape = NA,
               aes(fill = Treatment)) +
  geom_jitter(size = 1,
              alpha = 0.8,
              aes(color = Treatment)) +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  labs(title = "Perennial plant species \nrichness",
       x = NULL,
       y = "Coefficient of variation") +
  theme_bw() +
  theme(legend.position = "none") +
  scale_y_continuous(labels = percent) +
  theme(axis.text.x = element_text(color = "black")) +
  geom_text(aes(x = 0.95, y = 0.65, label = "Mann-Whitney, \np = 0.062"),
            color = "gray30",
            size = 2.5) +
  theme(plot.margin = margin(0.1, 0.1, 0.1, 0.1, "in")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
rich.plot.cv
```


## Supp Fig 5b: CV of perennial diversity
```{r, fig.height=4, fig.width=3}
# Shannon -----------------------------------------------------------------

# Find CV for each sample over time
shan.sample <- per.div |> 
  group_by(Sample, Treatment) |> 
  summarise(CV = sd(shan) / mean(shan),
            .groups = "keep")

# Compare
t.test(filter(shan.sample, Treatment == "Treated")$CV,
       filter(shan.sample, Treatment == "Control")$CV) # NS, p = 0.075

# Plot
shan.plot.cv <- shan.sample |> 
  ggplot(aes(x = Treatment, y = CV)) +
  geom_boxplot(alpha = 0.3,
               outlier.shape = NA,
               aes(fill = Treatment)) +
  geom_jitter(size = 1,
              alpha = 0.8,
              aes(color = Treatment)) +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  labs(title = "Perennial plant diversity",
       x = NULL,
       y = NULL) +
  theme_bw() +
  theme(legend.position = "none") +
  scale_y_continuous(labels = percent) +
  theme(axis.text.x = element_text(color = "black")) +
  geom_text(aes(x = 0.95, y = 0.83, label = "t-test, p = 0.075"),
            color = "gray30",
            size = 2.5) +
  theme(plot.margin = margin(0.1, 0.1, 0.1, 0.15, "in")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
shan.plot.cv
```


## Combine plots for Supp Fig 5
```{r, eval=FALSE}
# Combine richness & Shannon ----------------------------------------------

# Supplemental figure
tiff("figures/2023-09_publish-figures/CV_rich-shan.tiff", units = "in", height = 4, width = 5.5, res = 300)
ggarrange(rich.plot.cv, shan.plot.cv,
          ncol = 2, nrow = 1,
          labels = c("(A)", "(B)")) 

dev.off()
```


\newpage
# Appendix C: Soil fertility
## Setup
```{r, message=FALSE}
library(tidyverse)
library(ggpubr)
library(metagenomeSeq)
library(vegan)
```
```{r, eval=FALSE}
# Load data ---------------------------------------------------------------

barc.asv <- read.table("bac-arc_clean_asv.txt", sep = "\t",
                       header = T, row.names = 1)
fungi.asv <- read.table("fungi_clean_asv.txt",
                      sep = "\t", header = T, row.names = 1)
meta <- read.csv("sequencing_metadata.csv")

dat.2021 <- read.csv("Veg-soil-elev_2021.csv")
```
```{r}
# 16S ---------------------------------------------------------------------

# Normalization
barc.MR <- newMRexperiment(t(barc.asv))
p <- cumNormStat(barc.MR)
barc.MR <- cumNorm(barc.MR, p = p)
barc.norm <- t(MRcounts(barc.MR, norm = T, log = F))

# Richness and Shannon
meta$barc.richness <- specnumber(barc.norm)
meta$barc.shannon <- diversity(barc.norm, index = "shannon")

# Bray-Curtis distance
barc.dist <- vegdist(barc.norm, method = "bray")


# ITS ---------------------------------------------------------------------

# Normalization 
fungi.MR <- newMRexperiment(t(fungi.asv))
p <- cumNormStat(fungi.MR)
fungi.MR <- cumNorm(fungi.MR, p = p)
fungi.norm <- t(MRcounts(fungi.MR, norm = T, log = F))

# Richness and Shannon
meta$fungi.richness <- specnumber(fungi.norm)
meta$fungi.shannon <- diversity(fungi.norm, index = "shannon")

# Bray-Curtis distance
fungi.dist <- vegdist(fungi.norm, method = "bray")
```


## Fig 4a: Bacteria & archaea NMDS ordination
Note that PERMANOVA model results (`adonis2`) will vary slightly each time and will not exactly match values published in the paper.
```{r, fig.height=4, fig.width=4}
# NMDS ordination
barc.nmds <- metaMDS(barc.dist, k = 2)
barc.nmds$stress

meta$barc.NMDS1 <- barc.nmds$points[ , 1]
meta$barc.NMDS2 <- barc.nmds$points[ , 2]

# PERMANOVA
adonis2(barc.dist ~ meta$Treatment) 

# Plot (using dat.2021 for same NMDS values presented in paper):
barc.nmds.plot.21 <- dat.2021 %>% 
  ggplot(aes(x = barc.NMDS1, y = barc.NMDS2, color = Treatment, shape = Treatment)) +
  geom_point() +
  scale_color_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  labs(x = "Axis 1",
       y = "Axis 2",
       title = "Bacteria & archaea",
       color = "Treatment",
       shape = "Treatment") +
  theme(legend.position = "bottom") +
  theme(plot.margin = margin(0.1, 0.2, 0.1, 0.1, "in")) +
  theme(legend.title = element_blank()) +
  geom_text(aes(x = 0.2, y = -0.55, label = "PERMANOVA, p = 0.029"),
            size = 2.5, color = "gray30") +
  geom_text(aes(x = 0.25, y = -0.65, label = "Stress = 0.168"),
            size = 2.5, color = "gray30") +
  theme(plot.title = element_text(size = 12))
barc.nmds.plot.21
```

## Fig 4b: Fungi NMDS ordination
Note that PERMANOVA model results (`adonis2`) will vary slightly each time and will not exactly match values published in the paper.
```{r, fig.height=4, fig.width=4}
# NMDS ordination
fungi.nmds <- metaMDS(fungi.dist, k = 2)
fungi.nmds$stress

meta$fungi.NMDS1 <- fungi.nmds$points[ , 1]
meta$fungi.NMDS2 <- fungi.nmds$points[ , 2]

# PERMANOVA
adonis2(fungi.dist ~ meta$Treatment)


# Plot (using dat.2021 for same NMDS values presented in paper):
fungi.nmds.plot.21 <- dat.2021 %>% 
  ggplot(aes(x = fungi.NMDS1, y = fungi.NMDS2, color = Treatment, shape = Treatment)) +
  geom_point() +
  scale_color_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  labs(x = "Axis 1",
       y = "Axis 2",
       title = "Fungi",
       color = "Treatment",
       shape = "Treatment") +
  theme(legend.position = "bottom") +
  theme(legend.title = element_blank()) +
  theme(plot.margin = margin(0.1, 0.2, 0.1, 0.1, "in")) +
  geom_text(aes(x = 0.73, y = -0.7, label = "PERMANOVA, p = 0.011"),
            size = 2.5, color = "gray30") +
  geom_text(aes(x = 0.82, y = -0.8, label = "Stress = 0.236"),
            size = 2.5, color = "gray30") +
  theme(plot.title = element_text(size = 12))
fungi.nmds.plot.21
```

## Combine plots for Fig 4
```{r, eval=FALSE}
# Combine NMDS ------------------------------------------------------------

tiff("figures/2023-09_publish-figures/Soil-NMDS.tiff", height = 4, width = 7, units = "in", res = 1000)
ggarrange(barc.nmds.plot.21, fungi.nmds.plot.21,
          nrow = 1, ncol = 2,
          labels = c("(A)", "(B)"),
          common.legend = TRUE, legend = "bottom")
dev.off()
```


## Supp Fig 6a: Total soil nitrogen
```{r, fig.height=4, fig.width=3}
# Total N -----------------------------------------------------------------

# Mann-Whitney
wilcox.test(filter(dat.2021, Treatment == "Control")$TN_ppt,
            filter(dat.2021, Treatment == "Treated")$TN_ppt,
            exact = FALSE) # p-value = 0.1731

# Plot
tn.plot.21 <- dat.2021 |> 
  ggplot(aes(x = Treatment, y = TN_ppt)) +
  geom_boxplot(alpha = 0.3,
               outlier.shape = NA,
               aes(fill = Treatment)) +
  geom_jitter(size = 1,
              alpha = 0.8,
              aes(color = Treatment)) +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  labs(title = "Total soil nitrogen",
       x = NULL,
       y = "Total N (mg/g)") +
  theme_bw() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(color = "#000000"))  +
  theme(plot.margin = margin(0.1, 0.2, 0.1, 0.1, "in")) +
  geom_text(aes(x = 2, y = 63, label = "Mann-Whitney, \np = 0.173"),
            color = "gray30",
            size = 2.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
tn.plot.21
```

## Supp Fig 6b: Total soil carbon
```{r, fig.height=4, fig.width=3}
# Total C -----------------------------------------------------------------

# Mann-Whitney
wilcox.test(filter(dat.2021, Treatment == "Control")$TC_ppt,
            filter(dat.2021, Treatment == "Treated")$TC_ppt,
            exact = FALSE) # p-value = 0.2397

# Plot
tc.plot.21 <- dat.2021 |> 
  ggplot(aes(x = Treatment, y = TC_ppt)) +
  geom_boxplot(alpha = 0.3,
               outlier.shape = NA,
               aes(fill = Treatment)) +
  geom_jitter(size = 1,
              alpha = 0.8,
              aes(color = Treatment)) +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  labs(title = "Total soil carbon",
       x = NULL,
       y = "Total C (mg/g)") +
  theme_bw() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(color = "#000000"))  +
  theme(plot.margin = margin(0.1, 0.2, 0.1, 0.1, "in")) +
  geom_text(aes(x = 2, y = 670, label = "Mann-Whitney, \np = 0.240"),
            color = "gray30",
            size = 2.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
tc.plot.21
```


## Supp Fig 6c: Soil organic matter
```{r, fig.height=4, fig.width=3}
# Organic matter ----------------------------------------------------------

# Mann-Whitney
wilcox.test(filter(dat.2021, Treatment == "Control")$OM_perc,
            filter(dat.2021, Treatment == "Treated")$OM_perc) # p-value = 0.4332

# Plot
om.plot.21 <- dat.2021 |> 
  ggplot(aes(x = Treatment, y = OM_perc)) +
  geom_boxplot(alpha = 0.3,
               outlier.shape = NA,
               aes(fill = Treatment)) +
  geom_jitter(size = 1,
              alpha = 0.8,
              aes(color = Treatment)) +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  labs(title = "Soil organic matter",
       x = NULL,
       y = "Organic matter (%)") +
  theme_bw() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(color = "#000000"))  +
  theme(plot.margin = margin(0.1, 0.2, 0.1, 0.1, "in")) +
  geom_text(aes(x = 2, y = 2.14, label = "Mann-Whitney, \np = 0.433"),
            color = "gray30",
            size = 2.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
om.plot.21
```

## Combine plots for Supp Fig 6
```{r, eval=FALSE}
# Combine soil chem -------------------------------------------------------

# TN, TC, OM
tiff("figures/2023-09_publish-figures/Soil-chem.tiff", units = "in", height = 4, width = 7, res = 300)
ggarrange(tn.plot.21, tc.plot.21, om.plot.21, 
          ncol = 3, nrow = 1,
          labels = c("(A)", "(B)", "(C)")) 

dev.off()
```


## Supp Fig 7a: Bacterial & archaeal richness
```{r, fig.height=4, fig.width=3}
# Barc richness -----------------------------------------------------------

# T-test
t.test(filter(dat.2021, Treatment == "Control")$barc.richness,
       filter(dat.2021, Treatment == "Treated")$barc.richness) # NS, p = 0.686

# Plot
barc.rich.plot.21 <- dat.2021 |> 
  ggplot(aes(Treatment, barc.richness)) +
  geom_jitter(aes(color = Treatment), 
              alpha = 0.8, 
              size = 1) +
  geom_boxplot(aes(fill = Treatment), 
               alpha = 0.3, 
               outlier.shape = NA) +
  xlab(NULL) +
  ylab("No. of ASVs") +
  ggtitle("Bacterial & archaeal richness") +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(color = "#000000")) +
  geom_text(aes(x = 2.1, y = 1100, label = "t-test, p = 0.686"),
            color = "gray30",
            size = 2.5) +
  theme(plot.margin = margin(0.1, 0.1, 0.1, 0.1, "in")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 11.5))
barc.rich.plot.21
```


## Supp Fig 7b: Fungal richness
```{r, fig.height=4, fig.width=3}
# Fungi richness ----------------------------------------------------------

# T-test
t.test(filter(dat.2021, Treatment == "Control")$fungi.richness,
       filter(dat.2021, Treatment == "Treated")$fungi.richness) # NS, p = 0.938

# Plot
fungi.rich.plot.21 <- dat.2021 %>% 
  ggplot(aes(Treatment, fungi.richness)) +
  geom_jitter(aes(color = Treatment), 
              alpha = 0.8, 
              size = 1) +
  geom_boxplot(aes(fill = Treatment), 
               alpha = 0.3, 
               outlier.shape = NA) +
  xlab(NULL) +
  ylab(NULL) +
  ggtitle("Fungal richness") +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(color = "#000000")) +
  geom_text(aes(x = 2.1, y = 70, label = "t-test, p = 0.938"),
            color = "gray30",
            size = 2.5) +
  theme(plot.margin = margin(0.1, 0.1, 0.1, 0.1, "in")) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
fungi.rich.plot.21
```

## Combine plots for Supp Fig 7
```{r, eval=FALSE}
# Combine soil richness ---------------------------------------------------

tiff("figures/2023-09_publish-figures/Soil-richness.tiff", units = "in", height = 4, width = 5.5, res = 300)
ggarrange(barc.rich.plot.21, fungi.rich.plot.21,
          ncol = 2, nrow = 1,
          labels = c("(A)", "(B)"))

dev.off()
```


## Supp Fig 8a: Chemoheterotrophic bacteria & archaea
```{r, fig.height=4, fig.width=3}
# Chemoheterotrophs -------------------------------------------------------

# Mann-Whitney
wilcox.test(filter(dat.2021, Treatment == "Control")$chemoheterotrophy_perc,
            filter(dat.2021, Treatment == "Treated")$chemoheterotrophy_perc) # p = 0.8449

# Plot
chemohet.plot.21 <- dat.2021 %>% 
  ggplot(aes(Treatment, chemoheterotrophy_perc)) +
  geom_jitter(aes(color = Treatment), 
              alpha = 0.8, 
              size = 1) +
  geom_boxplot(aes(fill = Treatment), 
               alpha = 0.3, 
               outlier.shape = NA) +
  xlab(NULL) +
  ylab("Relative abundance (%)") +
  ggtitle("Chemoheterotrophic \nbacteria & archaea") +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(color = "#000000")) +
  theme(plot.margin = margin(0.1, 0.1, 0.1, 0.1, "in")) +
  geom_text(aes(x = 2, y = 28.7, label = "Mann-Whitney, \np = 0.845"),
            color = "gray30",
            size = 2.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
chemohet.plot.21
```


## Supp Fig 8b: Nitrogen-cycling bacteria & archaea
```{r, fig.height=4, fig.width=3}
# N-cyclers ---------------------------------------------------------------

# Mann-Whitney
wilcox.test(filter(dat.2021, Treatment == "Control")$n.cycler_perc,
            filter(dat.2021, Treatment == "Treated")$n.cycler_perc) # p-value = 0.5854

# Plot
ncycler.plot.21 <- dat.2021 %>% 
  ggplot(aes(Treatment, n.cycler_perc)) +
  geom_jitter(aes(color = Treatment), 
              alpha = 0.8, 
              size = 1) +
  geom_boxplot(aes(fill = Treatment), 
               alpha = 0.3, 
               outlier.shape = NA) +
  xlab(NULL) +
  ylab(NULL) +
  ggtitle("Nitrogen-cycling \nbacteria & archaea") +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(color = "#000000")) +
  theme(plot.margin = margin(0.1, 0.1, 0.1, 0.25, "in")) +
  geom_text(aes(x = 2, y = 13.6, label = "Mann-Whitney, \np = 0.585"),
            color = "gray30",
            size = 2.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
ncycler.plot.21
```


## Supp Fig 8c: Saprotrophic fungi
```{r, fig.height=4, fig.width=3}
# Saprotrophs -------------------------------------------------------------

# T-test
t.test(filter(dat.2021, Treatment == "Control")$saprotroph,
       filter(dat.2021, Treatment == "Treated")$saprotroph) # NS, p = 0.272

# Plot
sapro.plot.21 <- dat.2021 %>% 
  ggplot(aes(Treatment, saprotroph)) +
  geom_jitter(aes(color = Treatment), 
              alpha = 0.8, 
              size = 1) +
  geom_boxplot(aes(fill = Treatment), 
               alpha = 0.3, 
               outlier.shape = NA) +
  xlab(NULL) +
  ylab(NULL) +
  ggtitle("Saprotrophic fungi") +
  scale_color_manual(values = c("red", "#1F78B4")) +
  scale_fill_manual(values = c("red", "#1F78B4")) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(color = "#000000")) +
  theme(plot.margin = margin(0.1, 0.1, 0.1, 0.25, "in")) +
  geom_text(aes(x = 2, y = 42, label = "t-test, p = 0.272"),
            color = "gray30",
            size = 2.5) +
  stat_summary(fun = mean, geom = "errorbar", aes(ymax = after_stat(y), ymin = after_stat(y)),
               width = 0.75, linetype = "dashed") +
  theme(plot.title = element_text(size = 12))
sapro.plot.21
```

## Combine plots for Supp Fig 8
```{r}
# Combine FAPROTAX and FUNGuild -------------------------------------------

tiff("figures/Soil-functional.tiff", units = "in", height = 4, width = 7, res = 300)
ggarrange(chemohet.plot.21, ncycler.plot.21, sapro.plot.21, 
          ncol = 3, nrow = 1,
          labels = c("(A)", "(B)", "(C)")) 

dev.off()
```


# Appendix D: Structural equation modeling
## Setup
```{r, message=FALSE}
library(lavaan)
library(tidyverse)
```
```{r, eval=FALSE}
# Load data ---------------------------------------------------------------

dat.2021 <- read.csv("Veg-soil-elev_2021.csv") 
```


## Latent variables

## Model 1

## Model 2

## Model 3

## Model 4
